guides/source/routing.md

```diff
@@ -18,51 +18,66 @@ After reading this guide, you will know:
+helpers based on the resources configured in the router.
 
-### Connecting URLs to Code
+### Routing Incoming URLs to Code
 
-When your Rails application receives an incoming request for:
+When your Rails application receives an incoming request, it asks the router to match it to a controller action. For example, take the following incoming request:
 
 ```
-GET /patients/17
+GET /users/17
 ```
 
-it asks the router to match it to a controller action. If the first matching route is:
+The Router's job is to match it to a controller action. If the first matching route is:
 
 ```ruby
-get '/patients/:id', to: 'patients#show'
+get '/users/:id', to: 'user#show'
 ```
 
-the request is dispatched to the `patients` controller's `show` action with `{ id: '17' }` in `params`.
+The request is matched to the `UsersController` class's `show` action with `{ id: '17' }` in the `params` hash.
+
+The `to:` option expects a `controller#action` format when passed a string. Alternatively, You can pass a symbol and use the `action:` option, instead of `to:`. You can also pass a string without a `#`, in which case the `controller:` option is used instead to `to:`. For example:
+
+```ruby
+get '/users/:id', action: :show, controller: 'users'
+```
 
-NOTE: Rails uses snake_case for controller names here, if you have a multiple word controller like `MonsterTrucksController`, you want to use `monster_trucks#show` for example.
+NOTE: Rails uses snake_case for controller names when specifying routes. For example, if you have a controller named UserProfilesController, you would specify a route to the show action as user_profiles#show.
 
 ### Generating Paths and URLs from Code
 
-You can also generate paths and URLs. If the route above is modified to be:
+The Router automatically generates paths and URLs that can be used throughout your application, instead of hard-coded strings.
+
+For example, `user_path` and `user_url` helpers in the route below.
 
 ```ruby
-get '/patients/:id', to: 'patients#show', as: 'patient'
+get '/users/:id', to: 'users#show', as: 'user'
 ```
 
-and your application contains this code in the controller:
+NOTE: The `as:` option is used to provide a custom name for a route, which is used when generating URL and path helpers.
+
+Assuming your application contains this code in the controller:
 
 ```ruby
-@patient = Patient.find(params[:id])
+@user = User.find(params[:id])
 ```
 
 and this in the corresponding view:
 
 ```erb
-<%= link_to 'Patient Record', patient_path(@patient) %>
+<%= link_to 'User Record', user_path(@user) %>
 ```
 
-then the router will generate the path `/patients/17`. This reduces the brittleness of your view and makes your code easier to understand. Note that the id does not need to be specified in the route helper.
+The router will generate the path `/users/17` from `user_path(@user)`. Using the `user_path` helper allows you to avoid having to hard-code a path in your views.
```

Comment:
Does it make sense to explain _why_ not using hardcoded strings is helpful?
```suggestion
The router will generate the path `/users/17` from `user_path(@user)`. Using the `user_path` helper allows you to avoid having to hard-code a path in your views. This is helpful if you eventually move the route to a different URL, as you won't need to update the corresponding views.
```

Reply:
Sure!